[tensorexpr] Add Op handling for mobilenetv3 large

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64751
* #64750
* __->__ #64741
* #64740

Differential Revision: [D30839110](https://our.internmc.facebook.com/intern/diff/D30839110)

Adding 
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

Adding tensorexpr handling of 
prepacked::conv2d_clamp_run
prepacked::linear_clamp_run
to make ExternalCall that has prepacked custom class as a second argument.
This class will be put in constants as a void* and passed through bridge function.

Handling customClass is added to bindConstant

Handling 
aten::flatten as aten::view
aten::dropout as a noop tensor compute that will be eliminated in optimizations

The result:
mobilenetv3-large-optimized can be compiled to ll

IvanKobzarev committed:
[tensorexpr] Add Op handling for mobilenetv3 large

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839110).

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -1370,12 +1381,52 @@ Tensor computeConv2d(
   return Tensor(ResultBuf.node(), s);
 }
 
+Tensor computePrepackedConv2dClampRun(
```
Not for this PR, but it would nice to have all these `compute*` for external calls in a separate file, may be under `operators/` folder.

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -1370,12 +1381,52 @@ Tensor computeConv2d(
     const std::vector<ExprHandle>& outputShape,
     const c10::optional<ScalarType>& outputType,
     at::Device device) {
+  const std::string opStr = op.toQualString();
```
We could also move this logic into a separate function that lives in the file handling all external call stuff. Again, not blocking this diff, just for future cleanup.

navahgar reviewed (approved):
LGTM

Some minor comments to group the functions dealing with external calls in a separate file. Not blocking this diff though.

navahgar reviewed (commented):
Can you add tests for these ops?

IvanKobzarev committed:
Update on "[tensorexpr] Add Op handling for mobilenetv3 large"


Differential Revision: [D30839110](https://our.internmc.facebook.com/intern/diff/D30839110)

Adding 
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

Adding tensorexpr handling of 
prepacked::conv2d_clamp_run
prepacked::linear_clamp_run
to make ExternalCall that has prepacked custom class as a second argument.
This class will be put in constants as a void* and passed through bridge function.

Handling customClass is added to bindConstant

Handling 
aten::flatten as aten::view
aten::dropout as a noop tensor compute that will be eliminated in optimizations

The result:
mobilenetv3-large-optimized can be compiled to ll




[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839110).

IvanKobzarev committed:
Update on "[tensorexpr] Add Op handling for mobilenetv3 large"


Differential Revision: [D30839110](https://our.internmc.facebook.com/intern/diff/D30839110)

Adding 
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

Adding tensorexpr handling of 
prepacked::conv2d_clamp_run
prepacked::linear_clamp_run
to make ExternalCall that has prepacked custom class as a second argument.
This class will be put in constants as a void* and passed through bridge function.

Handling customClass is added to bindConstant

Handling 
aten::flatten as aten::view
aten::dropout as a noop tensor compute that will be eliminated in optimizations

The result:
mobilenetv3-large-optimized can be compiled to ll




[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839110).

IvanKobzarev committed:
Update on "[tensorexpr] Add Op handling for mobilenetv3 large"


Differential Revision: [D30839110](https://our.internmc.facebook.com/intern/diff/D30839110)

Adding 
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

Adding tensorexpr handling of 
prepacked::conv2d_clamp_run
prepacked::linear_clamp_run
to make ExternalCall that has prepacked custom class as a second argument.
This class will be put in constants as a void* and passed through bridge function.

Handling customClass is added to bindConstant

Handling 
aten::flatten as aten::view
aten::dropout as a noop tensor compute that will be eliminated in optimizations

The result:
mobilenetv3-large-optimized can be compiled to ll




[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839110).
